### ISSUE TITLE
Error: Client uploaded to the wrong offset

### ISSUE THREAD

**Issue by kxmatejka** [firebase/firebase-js-sdk#5125]
### [REQUIRED] Describe your environment

  * Operating System version: macOS 11 
  * Browser version: Safari 14
  * Firebase SDK version: 8.2.0
  * Firebase Product: storage

### [REQUIRED] Describe the problem

During the upload of multiple bigger files (video files of size 500-1000MB) some of the uploading files trow an error:
```
Uncaught (in promise) Error: FirebaseError:
Firebase Storage: An unknown error occurred, please check the error payload for server response. (storage/unknown)
Client uploaded to the wrong offset (872153088 instead of 891813888)
```

The invalid offset is either `838598659 instead of 869007360` or `872153088 instead of 891551744` but maybe I just don't have enough logs.

According to this debate https://groups.google.com/g/firebase-talk/c/ejc4hbwUNy4/m/PPPUN7rdAgAJ there was a similar issue on iOS where was a problem with calculation of blob size.

It looks like this problem appears only on safari.

#### Relevant Code:

There is an extract of logic for uploading files. It isn't anything fancy. Just asynchronous loop.

```javascript
import firebase from 'firebase/app'

const storageRef = getStorage().ref()

const uploadFile = async (
    file: File,
    filename: string,
    onProgress?: (progress: number) => void,
    metadata?: firebase.storage.SettableMetadata,
): Promise<string> => {
  const fileRef = storageRef.child(filename)
  const uploadTask = fileRef.put(file)

  uploadTask.on('state_changed', (snapshot) => {
    const progress = (snapshot.bytesTransferred / snapshot.totalBytes) * 100

    onProgress && onProgress(progress)
  })

  const snapshot = await uploadTask

  if (metadata) {
    try {
      await fileRef.updateMetadata(metadata)
    } catch (error) {
      loggerStorage.warn('File metadata update failed', filename, error, metadata)
    }
  }

  return snapshot.ref.getDownloadURL()
}

const URLs = await Promise.all(files.map((fileInterface) => uploadFile(fileInterface.file, fileInterface.filename))
```

Should we upload files synchronously? Or is there any other tip of what we should do differently?

**Comment by looptheloop88** [comment]
Hi @kxmatejka, thanks for the report. I wasn't able to reproduce the issue. I tried uploading > 1GB file multiple times using Safari and Google Chrome browsers, but all of them succeeded.

Have you tried updating the SDK version that you're using to the [latest](https://firebase.google.com/support/release-notes/js) one and see if the issue persists? By the way, I used the quickstart [sample project](https://github.com/firebase/quickstart-js/tree/master/storage) for Firebase Storage.
